x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java

```diff
@@ -683,4 +696,139 @@ private XContentBuilder getTokenIndexMappings() {
+    }
+
+    private Settings getProfileIndexSettings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1")
+            .put(IndexMetadata.SETTING_PRIORITY, 1000)
+            .put("index.refresh_interval", "1s")
+            .put(IndexMetadata.INDEX_FORMAT_SETTING.getKey(), INTERNAL_PROFILE_INDEX_FORMAT)
+            .build();
+    }
+
+    private XContentBuilder getProfileIndexMappings() {
+        try {
+            final XContentBuilder builder = jsonBuilder();
+            builder.startObject();
+            {
+                builder.startObject("_meta");
+                builder.field(SECURITY_VERSION_STRING, Version.CURRENT.toString());
+                builder.endObject();
+
+                builder.field("dynamic", "strict");
+                builder.startObject("properties");
+                {
+                    builder.startObject("uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("redirect_uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("enabled");
+                    builder.field("type", "boolean");
+                    builder.endObject();
+
+                    builder.startObject("user");
+                    {
+                        builder.field("type", "object");
+                        builder.startObject("properties");
+                        {
+                            builder.startObject("username");
+                            builder.field("type", "keyword");
+                            builder.endObject();
+
+                            builder.startObject("realm");
+                            {
+                                builder.field("type", "object");
+                                builder.startObject("properties");
+                                {
+                                    builder.startObject("name");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+
+                                    builder.startObject("type");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+
+                                    builder.startObject("domain");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+                                }
+                                builder.endObject();
+                            }
+                            builder.endObject();
+
+                            builder.startObject("email");
+                            builder.field("type", "text");
```

Comment:
There are examples in the main security index where we use the [UAX URL email tokenizer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-uaxurlemail-tokenizer.html) for this.

Reply:
Thanks for the pointer. I updated it to be the same as the email field of the main security index. I also wonder whether this could just be a keyword field because the tokenizer is more useful when the input is a text block. But what we have here for this field is likely just the email itself which does not require further analysis. That said, I like it being consistent with the settings used in the main security index.

Reply:
>  also wonder whether this could just be a keyword

I would lean to the `search-as-you-type` field for this too, because sometimes the email works as a name.
We don't have to settle it know, we should experiment with building the query for a multitude of such fields.

Reply:
I am not sure whether `search_as_you_type` is helpful here. Based on my understanding, it helps when the value has multiple tokens and you'd like to search these tokens in any order with prefix match. For an email, we can either treat it  as a single token or multiple tokens. In the case of single token, `search_as_you_type` does not really offer anything because there is only a single token (it might still be better at performance, not sure). 

In the case of multiple tokens, it may not make a lot sense because email `foo@example.com` maybe analyzed into `foo`, `example`, `com` and matching the later two tokens is unlikely useful.

In summary, I'll keep it as is for now and we can always come back to it later.